analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -125,7 +125,10 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
             descriptor.CustomTags.Contains(tag);
     }
 
-    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, string filePath) =>
+    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, SyntaxTree tree) =>
+        !IsUnchanged(tree) && IsInSonarLintContextAndIncluded(sonarLintXml, tree.FilePath);
+
+    private bool IsInSonarLintContextAndIncluded(SonarLintXmlReader sonarLintXml, string filePath) =>
         ProjectConfiguration().ProjectType != ProjectType.Unknown // Not SonarLint context, NuGet or Scanner <= 5.0
```

Comment:
This comment doesn't help much, instead of saying what it isn't it should say what it is. That will help organizing this method and naming it properly.
As a consequence, I'm not sure how to read it
```
Not SonarLint context || NuGet || Scanner <= 5.0
Not (SonarLint context || NuGet || Scanner <= 5.0)
```